desk/app/diary.hoon

```diff
@@ -156,6 +78,157 @@
+  |-
+  ?-  -.old
+      %2
+    =.  state  old
+    cor
+  ::
+      %1
+    %=  $
+      old  (state-1-to-2 old)
+    ==
+      %0
+    %=  $
+      old  (state-0-to-1 old)
+    ==
+  ==
+  ::
+  +$  versioned-state
+    $%  state-0
+        state-1
+        state-2
+    ==
+  ++  zero     zero:old:d
+  ++  one      unos:old:d
+  ++  two      d
+  +$  state-0  [%0 =shelf:zero]
+  +$  state-1  [%1 =shelf:one]
+  +$  state-2  current-state
+  ++  state-0-to-1
+    |=  sta=state-0
+    ^-  state-1
+    :-  %1
+    %-  ~(run by shelf.sta)
+    |=  =diary:zero
+    ^-  diary:one
+    %*  .  *diary:one
+      net    net.diary
+      log    (log-0-to-1 log.diary)
+      perm   perm.diary
+      view   view.diary
+      sort   sort.diary
+      notes  (notes-0-to-1 notes.diary banter.diary)
+      remark  remark.diary
+    ==
+  ::
+  ++  log-0-to-1
+    |=  =log:zero
+    ^-  log:one
+    %+  gas:log-on:one  *log:one
+    %+  turn  (tap:log-on:zero log)
+    |=  [=time =diff:zero]
+    ^-  [_time diff:one]
+    :-  time
+    ^-  diff:one
+    ?.  ?=(%quips -.diff)  diff
+    ^-  diff:one
+    [%notes p.diff %quips (quips-diff-0-to-1 q.diff)]
+  ::
+  ++  quips-diff-0-to-1
+    |=  =diff:quips:zero
+    ^-  diff:quips:one
+    :-  p.diff
+    ?.  ?=(%add -.q.diff)  q.diff
+    add/(memo-0-to-1 p.q.diff)
+  ::
+  ++  notes-0-to-1
+    |=  [=notes:zero banter=(map time quips:zero)]
+    ^-  notes:one
+    %+  gas:on:notes:one  *notes:one
+    %+  turn  (tap:on:notes:zero notes)
+    |=  [=time =note:zero]
+    ^-  [_time note:one]
+    :-  time
+    :_  +.note
+    ^-  seal:one
+    [time (quips-0-to-1 (~(gut by banter) time *quips:zero)) feels.note]
+  ::
+  ++  quips-0-to-1
+    |=  =quips:zero
+    ^-  quips:one
+    %+  gas:on:quips:one  *quips:one
+    ^-  (list [time quip:one])
+    %+  turn  (tap:on:quips:zero quips)
+    |=  [=time =quip:zero]
+    [time -.quip (memo-0-to-1 +.quip)]
+  ::
+  ++  memo-0-to-1
+    |=  =memo:zero
+    ^-  memo:one
+    [`content author sent]:memo
+  ::
+  ++  state-1-to-2
+    |=  sta=state-1
+    ^-  state-2
+    :-  %2
+    %-  ~(run by shelf.sta)
+    |=  =diary:one
+    ^-  diary:two
+    %*  .  *diary:two
+      net    (net-1-to-2 net.diary)
+      log    log.diary
+      perm   perm.diary
+      view   view.diary
+      sort   sort.diary
+      notes  notes.diary
+      remark  remark.diary
+    ==
+  ::
+  ++  net-1-to-2
+    |=  =net:one
+    ^-  net:two
+    ?.  ?=(%sub -.net)  net
+    [%sub p.net [%chi ~]]
+  --
+::
+++  watch-epic
+  |=  her=ship
+  ^+  cor
+  =/  =wire  /epic
+  =/  =dock  [her dap.bowl]
+  ?:  (~(has by wex.bowl) [wire dock])
+    cor
+  (emit %pass wire %agent [her dap.bowl] %watch /epic)
+::
+++  take-epic
+  |=  =sign:agent:gall
+  ^+  cor
+  ?+    -.sign  cor
+      %kick
+    ~&  'todo: check that sub is removed before ingesting kick'^wex.bowl
+    (watch-epic src.bowl)
+  ::
+      %fact
+    ?:  =(%epic p.cage.sign)
+      ~&  '!!! weird fact on /epic'
```

Comment:
note that not crashing here is correct, because crash paths in %fact handling can often lead to resubscribe loops and in the general will buy you more flexibility when you need to change the semantics.